modules/monitoring/main.tf

```diff
@@ -12,179 +12,99 @@ resource "azurerm_log_analytics_workspace" "log_analytics_workspace" {
+
+resource "azurerm_application_insights_smart_detection_rule" "server_response_time" {
+  name                        = "Degradation in server response time"
+  application_insights_id     = azurerm_application_insights.graphdb_insights.id
+  enabled                     = true
+  additional_email_recipients = var.email_recipients
 }
 
-resource "azurerm_monitor_data_collection_rule" "data_collection_rule" {
-  name                        = "dcr-${var.resource_group_name}"
-  resource_group_name         = var.resource_group_name
-  location                    = var.location
-  data_collection_endpoint_id = azurerm_monitor_data_collection_endpoint.data_collection_endpoint.id
-  kind                        = var.data_collection_kind
-
-  destinations {
-    log_analytics {
-      workspace_resource_id = azurerm_log_analytics_workspace.log_analytics_workspace.id
-      name                  = local.log_analytics_destination_name
-    }
-  }
+resource "azurerm_application_insights_smart_detection_rule" "rise_exception_volume" {
+  name                        = "Abnormal rise in exception volume"
+  application_insights_id     = azurerm_application_insights.graphdb_insights.id
+  enabled                     = true
+  additional_email_recipients = var.email_recipients
+}
 
-  data_flow {
-    streams      = ["Microsoft-Syslog", "Microsoft-Perf", "Microsoft-InsightsMetrics"]
-    destinations = [local.log_analytics_destination_name]
-  }
+resource "azurerm_application_insights_smart_detection_rule" "memory_leak_detection" {
+  name                        = "Potential memory leak detected"
+  application_insights_id     = azurerm_application_insights.graphdb_insights.id
+  enabled                     = true
+  additional_email_recipients = var.email_recipients
+}
 
-  data_flow {
-    streams       = ["Custom-${azapi_resource.main_gdb_log_table.name}"]
-    destinations  = [local.log_analytics_destination_name]
-    output_stream = "Custom-${azapi_resource.main_gdb_log_table.name}"
-    transform_kql = "source"
-  }
+resource "azurerm_application_insights_smart_detection_rule" "security_issue_detection" {
+  name                        = "Potential security issue detected"
+  application_insights_id     = azurerm_application_insights.graphdb_insights.id
+  enabled                     = true
+  additional_email_recipients = var.email_recipients
+}
 
-  data_sources {
-    syslog {
-      facility_names = ["*"]
-      log_levels     = ["*"]
-      name           = "gdb-datasource-syslog"
-      streams        = ["Microsoft-Syslog"]
-    }
+resource "azurerm_application_insights_smart_detection_rule" "rise_daily_data_volume" {
+  name                        = "Abnormal rise in daily data volume"
+  application_insights_id     = azurerm_application_insights.graphdb_insights.id
+  enabled                     = true
+  additional_email_recipients = var.email_recipients
+}
 
-    log_file {
-      name          = local.gdb_main_log_table_name
```

Comment:
gdb_main_log_table_name is now a dangling local

Reply:
removed